src/controller/board.py

```
@@ -86,37 +82,38 @@ def update_board(self, id: int, current_user: User, body: BodyUpdateBoard) -> No
-        self.session.delete(board)
         try:
-            self.session.commit()
+            self.board_repository.delete(board)
         except IntegrityError:
-            self.session.rollback()
             logger.warning(
                 f"err_type=board_delete err_code=409 ; msg=Cannot delete board because of foreign key restriction ; board_id={id} ; executor={current_user.id}"
             )
             raise HTTPException(
                 409, detail="Cannot delete board because of foreign key restriction"
             )
```

Ruff (0.14.5)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)